feature-libs/product-configurator/rulebased/components/attribute/types/base/configurator-attribute-base.component.ts

```diff
@@ -358,4 +396,58 @@ export class ConfiguratorAttributeBaseComponent {
       !this.isReadOnly(attribute)
     );
   }
+
+  /**
+   * Merges value price data received via @see {ConfiguratorValuePriceChanged} events into the given value, if available.
+   * @param {Configurator.Value} value the value
+   * @returns {Configurator.Value} the new value with pricing
+   */
+  protected mergePriceAndValue(value: Configurator.Value): Configurator.Value {
+    const valueName = value.name;
+    if (valueName && this.valuePrices[valueName]) {
+      value = { ...value, valuePrice: this.valuePrices[valueName] };
+    }
+    return value;
+  }
+
+  /**
+   * Creates a text describing the current attribute that can be used as ARIA label.
+   * Includes price information. If a total price is available this price will be used,
+   * otherwise it falls back to the value price, or if no price is available,
+   * no price information will be included in the text.
+   *
+   * @param {Configurator.Attribute} attribute the attribute
+   * @param {Configurator.Value} value the value
+   * @returns {string} translated text
+   */
+  protected getAriaLabelGeneric(
```

Comment:
yes - because some inheriting components already define a `getAriaLabel `method, but with different signature (order of params!):

- configurator-attribute-read-only.component.ts` getAriaLabel(value: Configurator.Value, attribute: Configurator.Attribute)`
- configurator-attribute-single-selection-base.component.ts` getAriaLabel(attribute: Configurator.Attribute, value: Configurator.Value)`
hence it is impossible to define a 'genric' `getAriaLabel ` method without a breaking change.

Reply:
@Uli-Tiger, 
what about to deprecate the existing method in `configurator-attribute-single-selection-base.component.ts` due to another order of the parameters and to override the implementation in:
- configurator-attribute-read-only.component.ts  and
- configurator-attribute-single-selection-base.component.ts  components.

Then you could adjust the method name to `getAriaLabel` in `configurator-attribute-base.component.ts`  and call `getAriaLabel`  in the HTML tree accordingly.

Reply:
@Uli-Tiger, could you have a look at unresolved comments? 
Thanks a lot in advance!